src/Spice86.Core/Emulator/VM/EmulationLoop.cs

```diff
@@ -99,7 +86,7 @@ public class EmulationLoop {
     private void RunLoop() {
         _stopwatch.Start();
         while (_cpuState.IsRunning) {
-            PauseIfAskedTo();
+            _pauseHandler.WaitIfPaused();
```

Comment:
Just noticed WaitIfPaused is called here but also in _machineBreakpoints.CheckBreakPoint()
This is not due to your change, but while you are at it, I wonder if this is not someting that could be removed either here on in the MachineBreakpoints class.

Reply:
I dropped the one from CheckBreakPoint and reorderd the calls so that if checking the breakpoint causes a pause, it is triggered immediately